extensions/ql-vscode/src/vscode-tests/factories/remote-queries/shared/variant-analysis.ts

```diff
@@ -0,0 +1,34 @@
+import {
+  VariantAnalysis,
+  VariantAnalysisQueryLanguage,
+  VariantAnalysisScannedRepository,
+  VariantAnalysisSkippedRepositories,
+  VariantAnalysisStatus,
+} from '../../../../remote-queries/shared/variant-analysis';
+import { createMockScannedRepos } from './scanned-repositories';
+import { createMockSkippedRepos } from './skipped-repositories';
+
+export function createMockVariantAnalysis(
+  status: VariantAnalysisStatus = VariantAnalysisStatus.InProgress,
+  scannedRepos: VariantAnalysisScannedRepository[] = createMockScannedRepos(),
+  skippedRepos: VariantAnalysisSkippedRepositories = createMockSkippedRepos()
+): VariantAnalysis {
+  const variantAnalysis: VariantAnalysis = {
+    id: 123,
```

Comment:
Would it make sense to also use `faker.datatype.number()` here (and for the other IDs) like in the repository mock?

Reply:
Makes sense to me. Have added it [here](https://github.com/github/vscode-codeql/pull/1559/commits/f56f017a8401b56d4a45971f7ad03a52d4ac5472).